src/models/ctx/update_notifications.rs

```diff
@@ -0,0 +1,213 @@
+            match notifications.items.remove(id) {
+                Some(_) => Effects::msg(Msg::Internal(Internal::NotificationsChanged)).unchanged(),
+                _ => Effects::none().unchanged(),
+            }
+        }
+        Msg::Internal(Internal::NotificationsChanged) => {
+            Effects::one(push_notifications_to_storage::<E>(notifications)).unchanged()
+        }
+        _ => Effects::none().unchanged(),
+    }
+}
+
+fn update_notification_items(
+    notification_items: &mut HashMap<String, NotificationItem>,
+    notification_catalogs: &Vec<ResourceLoadable<Vec<MetaItem>>>,
+    library: &LibraryBucket,
+) -> Effects {
+    let selected_catalogs = notification_catalogs
+        .iter()
+        .take_while(|catalog| match &catalog.content {
+            Some(Loadable::Ready(_)) | Some(Loadable::Err(_)) => true,
+            _ => false,
+        })
+        .collect::<Vec<_>>();
+    let next_notification_ids = notification_catalogs
+        .first()
+        .map(|resource| &resource.request.path.extra)
+        .map(|extra| Either::Left(extra.iter()))
+        .unwrap_or_else(|| Either::Right(std::iter::empty()))
+        .find(|extra_value| extra_value.name == LAST_VIDEOS_IDS_EXTRA_PROP.name)
+        .map(|extra_value| Either::Left(extra_value.value.split(",")))
+        .unwrap_or_else(|| Either::Right(std::iter::empty()));
+    let next_notification_items = next_notification_ids
+        .filter_map(|id| {
+            if let Some(notification_item) = notification_items.get(id) {
+                return Some(notification_item.to_owned());
+            };
+
+            let library_item = library.items.get(id);
+            let meta_item = selected_catalogs.iter().find_map(|catalog| {
+                catalog
+                    .content
+                    .as_ref()
+                    .and_then(|content| content.ready())
+                    .and_then(|content| {
+                        content.iter().find(|meta_item| meta_item.preview.id == *id)
+                    })
+            });
+            match (library_item, meta_item) {
+                (Some(library_item), Some(meta_item))
+                    if library_item.should_pull_notifications() && !meta_item.videos.is_empty() =>
+                {
+                    meta_item
+                        .videos_iter()
+                        .find(|video| {
+                            match (&library_item.state.last_video_released, &video.released) {
+                                (Some(last_video_released), Some(video_released)) => {
+                                    video_released > last_video_released
+                                }
+                                _ => false,
+                            }
+                        })
+                        .map(|video| NotificationItem {
+                            id: id.to_owned(),
+                            video: video.to_owned(),
+                        })
```

Comment:
Having a single video for a given LIbraryItem could be a problem because we might end up showing a number with the new episodes.